Sources/_CryptoExtras/MLDSA/MLDSA_boring.swift

```diff
@@ -0,0 +1,352 @@
+
+        fileprivate final class Backing {
+            private let pointer: UnsafeMutablePointer<MLDSA65_private_key>
+
+            func withUnsafePointer<T>(_ body: (UnsafePointer<MLDSA65_private_key>) throws -> T) rethrows -> T {
+                try body(self.pointer)
+            }
+
+            /// Initialize a ML-DSA-65 private key from a random seed.
+            init() throws {
+                self.pointer = UnsafeMutablePointer<MLDSA65_private_key>.allocate(capacity: 1)
+
+                try withUnsafeTemporaryAllocation(of: UInt8.self, capacity: MLDSA.PublicKey.Backing.bytesCount) { publicKeyPtr in
+                    try withUnsafeTemporaryAllocation(of: UInt8.self, capacity: MLDSA.seedSizeInBytes) { seedPtr in
+                        guard CCryptoBoringSSL_MLDSA65_generate_key(
+                            publicKeyPtr.baseAddress,
+                            seedPtr.baseAddress,
+                            self.pointer
+                        ) == 1 else {
+                            throw CryptoKitError.internalBoringSSLError()
+                        }
+                    }
+                }
+            }
+
+            /// Initialize a ML-DSA-65 private key from a seed.
+            /// 
+            /// The seed must be at least 32 bytes long.
+            /// Any additional bytes in the seed are ignored.
+            /// 
+            /// - Parameter seed: The seed to use to generate the private key.
+            /// 
+            /// - Throws: `CryptoKitError.incorrectKeySize` if the seed is not at least 32 bytes long.
+            init(seed: some DataProtocol) throws {
+                guard seed.count >= MLDSA.seedSizeInBytes else {
+                    throw CryptoKitError.incorrectKeySize
+                }
+
+                self.pointer = UnsafeMutablePointer<MLDSA65_private_key>.allocate(capacity: 1)
+
+                guard CCryptoBoringSSL_MLDSA65_private_key_from_seed(
+                    self.pointer,
+                    Array(seed.prefix(MLDSA.seedSizeInBytes)),
+                    MLDSA.seedSizeInBytes
+                ) == 1 else {
+                    throw CryptoKitError.internalBoringSSLError()
+                }
+            }
+
+            /// Initialize a ML-DSA-65 private key from a raw representation.
+            /// 
+            /// - Parameter rawRepresentation: The private key bytes.
+            /// 
+            /// - Throws: `CryptoKitError.incorrectKeySize` if the raw representation is not the correct size.
+            init(rawRepresentation: some DataProtocol) throws {
+                guard rawRepresentation.count == MLDSA.PrivateKey.Backing.bytesCount else {
+                    throw CryptoKitError.incorrectKeySize
+                }
+
+                self.pointer = UnsafeMutablePointer<MLDSA65_private_key>.allocate(capacity: 1)
+
+                try rawRepresentation.regions.flatMap { $0 }.withUnsafeBufferPointer { buffer in
```

Comment:
When I rewrite it like
```swift
let bytes: ContiguousBytes = rawRepresentation.regions.count == 1 ? rawRepresentation.regions.first! : Array(rawRepresentation)
try bytes.withUnsafeBytes { buffer in
    var cbs = CBS(data: buffer.baseAddress, len: buffer.count)
    guard CCryptoBoringSSL_MLDSA65_parse_private_key(self.pointer, &cbs) == 1 else {
        throw CryptoKitError.internalBoringSSLError()
    }
}
```
an error on the CBS init appears, on `buffer.baseAddress`: `Cannot convert value of type 'UnsafeRawPointer?' to expected argument type 'UnsafePointer<UInt8>?' because initializer 'init(data:len:)' was not imported from C header`.

Changing it to `CBS(data: buffer.baseAddress?.assumingMemoryBound(to: UInt8.self), len: buffer.count)` seems to work, is it fine?